src/main/kotlin/com/github/avrokotlin/avro4k/encoder/GenericAvroEncoder.kt

```diff
@@ -0,0 +1,22 @@
+package com.github.avrokotlin.avro4k.encoder
+
+import com.github.avrokotlin.avro4k.AvroConfiguration
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.modules.SerializersModule
+import org.apache.avro.Schema
+
+@ExperimentalSerializationApi
+class GenericAvroEncoder(
+        schema: Schema,
+        override val serializersModule: SerializersModule,
+        override val configuration: AvroConfiguration,
+) : AbstractAvroEncoder() {
+    var encodedValue: Any? = null
+        private set
+
+    override fun encodeNativeValue(value: Any?) {
+        encodedValue = value
+    }
+
+    override val currentUnresolvedSchema: Schema = schema
+}
```

Comment:
Why I prefer to not split the not-only-records in root ? This class, it seems easy, but needs the rest to work 😄